data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/util/LiteralMatcher.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression.util;
+
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.hamcrest.Description;
+import org.hamcrest.DiagnosingMatcher;
+import org.hamcrest.Matcher;
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import static org.opensearch.dataprepper.expression.util.ContextMatcher.hasContext;
+import static org.opensearch.dataprepper.expression.util.TerminalNodeMatcher.isTerminalNode;
+
+public class LiteralMatcher extends SimpleExpressionMatcher {
+    private static final Matcher<ParseTree> LITERAL_MATCHER = hasContext(DataPrepperExpressionParser.LiteralContext.class, isTerminalNode());
+
+    //region valid rule order
+    private static final RuleClassOrderedList VALID_LITERAL_RULE_ORDER = new RuleClassOrderedList(
+            DataPrepperExpressionParser.ExpressionContext.class,
+            DataPrepperExpressionParser.ConditionalExpressionContext.class,
+            DataPrepperExpressionParser.EqualityOperatorExpressionContext.class,
+            DataPrepperExpressionParser.RegexOperatorExpressionContext.class,
+            DataPrepperExpressionParser.RelationalOperatorExpressionContext.class,
+            DataPrepperExpressionParser.SetOperatorExpressionContext.class,
+            DataPrepperExpressionParser.UnaryOperatorExpressionContext.class,
+            DataPrepperExpressionParser.PrimaryContext.class,
+            DataPrepperExpressionParser.LiteralContext.class
+    );
+    //endregion
+
+    protected LiteralMatcher(final RuleClassOrderedList validRuleOrder) {
+        super(validRuleOrder);
+    }
+
+    /**
+     * Creates a matcher to check if a nodes is a unary tree with all nodes in a valid order that ends in a terminal node
+     * @return DiagnosingMatcher
+     *
+     * @see TerminalNodeMatcher#isTerminalNode()
+     */
+    public static DiagnosingMatcher<ParseTree> isUnaryTree() {
```

Comment:
nit: since this method is returning a matcher and return type is not boolean, consider adding verb prefix to the method, may be createUnaryTreeMatcher.

Reply:
Thanks for the suggestion @dinujoh! isUnaryTree() will be used similar to Hamcrest's `isEqual()`.

A quick example:
```
assertThat(item, isEqual(expected));
assertThat(item, isUnaryTree());
```

Using something like createUnaryTreeMatcher reads strange:
```
assertThat(item, createUnaryTreeMatcher());
```

I'm open to naming suggestions if `isUnaryTree()` is not descriptive.

Reply:
Makes sense from Matcher point of view. Thanks for clarifying.